lib/node_modules/@stdlib/blas/ext/base/scusum/src/addon.c

```diff
@@ -0,0 +1,51 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2020 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/blas/ext/base/scusum.h"
+#include "stdlib/napi/export.h"
+#include "stdlib/napi/argv.h"
+#include "stdlib/napi/argv_int64.h"
+#include "stdlib/napi/argv_float.h"
+#include "stdlib/napi/argv_strided_float32array.h"
+#include <node_api.h>
+#include <assert.h>
+
+/**
+* Receives JavaScript callback invocation data.
+*
+* @private
+* @param env    environment under which the function is invoked
+* @param info   callback data
+* @return       Node-API value
+*/
+static napi_value addon( napi_env env, napi_callback_info info ) {
+	STDLIB_NAPI_ARGV( env, info, argv, argc, 6 );
+	STDLIB_NAPI_ARGV_INT64( env, N, argv, 0 );
+	STDLIB_NAPI_ARGV_FLOAT( env, N, argv, 1 );
+	STDLIB_NAPI_ARGV_INT64( env, strideX, argv, 3 );
+	STDLIB_NAPI_ARGV_INT64( env, strideY, argv, 5 );
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT32ARRAY( env, X, N, strideX, argv, 2 );
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT32ARRAY( env, X, N, strideY, argv, 4 );
+
+	napi_value v;
+	status = napi_create_double( env, stdlib_strided_scusum( N, sum, X, strideX, Y, strideY ), &v );
+	assert( status == napi_ok );
+	return v;
```

Comment:
Wait...sorry. I missed this in `dcusum`. What is the return value of `stdlib_strided_scusum`? I believe we aren't returning any value. And in the `addon.cpp`, we returned a `nullptr` (see https://github.com/stdlib-js/stdlib/pull/1864/files#diff-9e113d4e7147f5a08bf0703c64a82b428dcfa4b9468802df73ad1d4e3cc1d73aL152).

As such, this implementation is incorrect.

Reply:
For an example of what I'd expect, see https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/blas/ext/base/dfill/src/addon.c.

You can also open up a patch for `dcusum` with the requested changes.

Reply:
I will soon fix this and open the patch also.

Reply:
Great! Thanks!